### PR TITLE
Add support for multiple attributes per HTML element

### PR DESCRIPTION
**Pull request by MotoLegacy**
Also fixes a bug with boolean attributes, and includes a modified test case.

### CODE REVIEW

File: source/html_attributeparser.c [hunk 1]
```diff
@@ -1215,45 +1215,69 @@ valuetype_t HTML_ParseValueTypeAttribute(char* value)
...
+void HTML_ParseAttributeContent(char* html_data, char* attribute_name, bool has_value,
                                         attribute_t* attributes, int* offset)
 {
     // First off - we need to store the value in a char*,
-    // so let's do that.
+    // so let's prep that.
+    char* attr_value = NULL;
+    char* real_value = NULL;
+
+    if (has_value == true) {
```
Comment (krazynez) [inline source/html_attributeparser.c]: if (has_value)

File: source/html_attributeparser.c [hunk 2]
```diff
@@ -1215,45 +1215,69 @@ valuetype_t HTML_ParseValueTypeAttribute(char* value)
...
+        // Store the offset index for the value parsing.
+        int val_offset = 0;
 
-        // Otherwise, append to value contents.
-        attr_value[val_offset] = html_data[i + *offset];
-        attr_value[val_offset + 1] = '\0';
-        val_offset++;
-    }
+        // Just like BeginParse, iterate through the document.
+        int i = 0;
```
Comment (krazynez) [inline source/html_attributeparser.c:1251]: this is old practice should be in the for loop (try running with clang)

**Review by krazynez — COMMENTED** [review]
Non function breaking things its okay for approval just avoid warnings.

File: source/html_attributeparser.c (hunk 2 above)
Comment (MotoLegacy) [reply source/html_attributeparser.c:1251]: `i` is referenced outside of the loop, therefore should not be delegated to initiation inside of it.

**Pull request merged**